core/src/main/scala/com/microsoft/azure/synapse/ml/train/TrainClassifier.scala

```diff
@@ -162,18 +164,34 @@ class TrainClassifier(override val uid: String) extends AutoTrainer[TrainedClass
         convertedLabelDataset.columns.filterNot(nonFeatureColumns.contains)
       }
 
+      println(s"$this - [trainClassifier] start featurize at ${Calendar.getInstance().getTime()}")
       val featurizer = new Featurize()
         .setOutputCol(getFeaturesCol)
         .setInputCols(featureColumns)
         .setOneHotEncodeCategoricals(oneHotEncodeCategoricals)
         .setNumFeatures(featuresToHashTo)
       val featurizedModel = featurizer.fit(convertedLabelDataset)
-      val processedData = featurizedModel.transform(convertedLabelDataset)
 
-      processedData.cache()
+// Error: java.util.NoSuchElementException: Failed to find a default value for weightCol
+//      val colstoSelect = if (isDefined(weightCol) || !$(weightCol).isEmpty)
+//        Array(getFeaturesCol, getLabelCol, getWeightCol)
+//      else Array(getFeaturesCol, getLabelCol)
+//
+//      val processedData = featurizedModel.transform(convertedLabelDataset).select(colstoSelect.map(col): _*)
+
+      val processedData = featurizedModel.transform(convertedLabelDataset).select(getFeaturesCol, getLabelCol)
```

Comment:
This line of the code is to unblock the cache and fit afterwards. 

There is a performance issue here, it will cache the whole dataset, but we only need the "features" and "label" column.

Reply:
But the change will make another problem, if user set WeightCol, it will throw exception that weightCol is not available. 
I tried to add it above (the commented code) but it's not working, 

if (isDefined(weightCol) || !$(weightCol).isEmpty)
this will throw exception, java.util.NoSuchElementException: Failed to find a default value for weightCol